app/src/main/java/id_iot/demo/CollectionAdapter.java

```diff
@@ -0,0 +1,102 @@
+package id_iot.demo;
+
+import android.content.Context;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Filterable;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Filter;
+import android.widget.Filterable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CollectionAdapter extends BaseAdapter implements Filterable {
+    private Context mContext;
```

Comment:
you use "m" prefix here as Android convention says, but rest of your fields are missing this. So this makes your convention inconsistent